### PR TITLE
refactor: Split service.Handle function for Inbound/Outbound type

### LINKED ISSUE #490: Split service.Handle function for Inbound/Outbound type
The [service.Handle](https://github.com/hyperledger/aries-framework-go/blob/master/pkg/didcomm/protocol/didexchange/service.go#L112) function handles both inbound messages from dispatcher as well as outbound flow as described in #500.  

The service should have below APIs.
- HandleInbound(DIDCommMsg)
- HandleOutbound(DIDCommMsg, Destination)

### PR DESCRIPTION
**Pull request by rolsonquadras**
- Add following APIs to handle the messages in service layer.
	- HandleInbound(DIDCommMsg)
	- HandleOutbound(DIDCommMsg, Destination)
- Remove outbound related fields from DIDCommMsg struct

Closes #490 

Signed-off-by: Rolson Quadras <rolson.quadras@securekey.com>

### CODE REVIEW

File: pkg/didcomm/common/service/service.go [hunk 1]
```diff
@@ -20,16 +20,24 @@ type DIDComm interface {
...
-// DIDCommMsg did comm msg
+// DIDCommMsg is used by handlers/dispatchers/events to pass DIDComm messages.
 type DIDCommMsg struct {
+	// message type
+	Type string
+	// message
+	Payload []byte
+}
+
+// Msg is used for outbound related features.
```
Comment (troyronda) [inline pkg/didcomm/common/service/service.go]: There is a boolean inside this struct that could mean incoming?

File: pkg/didcomm/protocol/didexchange/service.go [hunk 2]
```diff
@@ -112,7 +112,7 @@ func (s *Service) Handle(msg *service.DIDCommMsg) error {
 
 	logger.Infof("entered into Handle exchange message : %s", msg.Payload)
 
-	if !msg.Outbound && aEvent == nil {
```
Comment (troyronda) [inline pkg/didcomm/protocol/didexchange/service.go]: was the msg.Outbound implied?

File: pkg/didcomm/common/service/service.go (hunk 1 above)
Comment (rolsonquadras) [reply pkg/didcomm/common/service/service.go]: The boolean is not set anywhere outside the service layer. It look to manipulate the state machine operations. e.r. [here](https://github.com/hyperledger/aries-framework-go/blob/master/pkg/didcomm/protocol/didexchange/states.go#L218).

Comment (rolsonquadras) [reply pkg/didcomm/common/service/service.go]: [Updated](https://github.com/hyperledger/aries-framework-go/compare/0b5ffafa1afbb9931651ae126eeaf822901dda9f..5826f9a5bccaf6a6bfdbe2bfa90f56f34194147e#diff-75342fbf986feea5a6bb6a33d0caff76R31) the comment (added state machine).

File: pkg/didcomm/protocol/didexchange/service.go (hunk 2 above)
Comment (rolsonquadras) [reply pkg/didcomm/protocol/didexchange/service.go]: Outbound flag is used in state machine. The inbound dispatcher never sets this (always false) https://github.com/hyperledger/aries-framework-go/pull/482#discussion_r334157477.

File: pkg/didcomm/common/service/service.go [hunk 3]
```diff
@@ -20,16 +20,24 @@ type DIDComm interface {
...
+// DIDCommMsg is used by handlers/dispatchers/events to pass DIDComm messages.
 type DIDCommMsg struct {
+	// message type
+	Type string
+	// message
+	Payload []byte
+}
+
+// Msg is used for state machine processing.
+type Msg struct {
```
Comment (troyronda) [inline pkg/didcomm/common/service/service.go]: Perhaps we should have a TODO for splitting inbound and outbound type.

File: pkg/didcomm/protocol/didexchange/service.go (hunk 2 above)
Comment (troyronda) [reply pkg/didcomm/protocol/didexchange/service.go]: Hmmm. I think the way this code is being refactored implies that the same type shouldn’t be used... commented above.

File: pkg/didcomm/common/service/service.go (hunk 3 above)
Comment (rolsonquadras) [reply pkg/didcomm/common/service/service.go]: Created an issue #490 and added a [TODO](https://github.com/hyperledger/aries-framework-go/pull/482/files#diff-75342fbf986feea5a6bb6a33d0caff76R32).

Comment (rolsonquadras) [reply pkg/didcomm/common/service/service.go]: Updated the changes in this PR.

File: pkg/didcomm/protocol/didexchange/service.go [hunk 4]
```diff
@@ -69,6 +69,14 @@ type connectionStore interface {
 	GetConnection(connectionID string) (*ConnectionRecord, error)
 }
 
+// stateMachineMsg is an internal struct used to pass data to state machine.
+type stateMachineMsg struct {
```
Comment (troyronda) [inline pkg/didcomm/protocol/didexchange/service.go]: are we planning to keep a combined inbound/outbound structure within the state machine?

Comment (rolsonquadras) [reply pkg/didcomm/protocol/didexchange/service.go]: We can separate [Execute](https://github.com/hyperledger/aries-framework-go/blob/master/pkg/didcomm/protocol/didexchange/states.go#L54) function into ExecuteInbound and ExecuteOutbound.

Comment (rolsonquadras) [reply pkg/didcomm/protocol/didexchange/service.go]: Created an issue - https://github.com/hyperledger/aries-framework-go/issues/501

File: pkg/didcomm/common/service/service.go [hunk 5]
```diff
@@ -15,7 +15,8 @@ import (
 
 // Handler provides protocol service handle api.
 type Handler interface {
-	Handle(msg *DIDCommMsg) error
+	// HandleInbound handles inbound didexchange messages.
+	HandleInbound(msg *DIDCommMsg) error
```
Comment (llorllale) [inline pkg/didcomm/common/service/service.go]: HandleOutbound should've been added to this interface as well.

**Pull request merged**